src/main/java/org/opensearch/knn/quantization/models/quantizationState/QuantizationStateCache.java

```diff
@@ -71,6 +78,8 @@ private void buildCache() {
             )
             .removalListener(this::onRemoval)
             .build();
+
+        this.cacheMaintainer = new ScheduledExecutor(() -> cache.cleanUp(), 60 * 1000);
```

Comment:
Could you consider to use `KNN_CACHE_ITEM_EXPIRY_TIME_MINUTES`?
By using the setting value, we can expect that max retention time after time expiration will be `KNN_CACHE_ITEM_EXPIRY_TIME_MINUTES` minutes, but average retention time would be half of it.

Reply:
Also could you wrap `cache.cleanUp` with try-catch to make sure not throwing an exception?
Once it throws an exception, it will never be rescheduled.